### PR TITLE
Change method scope to support extending module classes

### PR DESCRIPTION
**Pull request by karenhanson**
Portico is working to upgrade to 1.23 - we currently use a modified older version of JHOVE. There are several method/property scope changes would that allow us to continue to use some local module extensions without having to maintain a customized version of JHOVE. If they are benign (or even useful) for other users, it would be great if we could merge these.  

Question for reviewer(s): one of the changes is in the `xml-hul` module. Should I increment the version? and if so to a SNAPSHOT version?

Thanks!

### CODE REVIEW

File: jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java [hunk 1]
```diff
@@ -897,10 +897,15 @@ else if (arity.equals (PropertyArity.ARRAY)) {
...
+    /**
+     * Checks if a property would produce an empty XML element, and returns true if
+     * it would.
+     * 
+     * @param property
+     * @param arity
+     * @return
      */
-    private boolean isPropertyEmpty (Property property, PropertyArity arity) {
+    public boolean isPropertyEmpty (Property property, PropertyArity arity) {
```
Comment (tledoux) [inline jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java:908]: Shouldn't this method be static also ?

File: jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java [hunk 2]
```diff
@@ -4406,10 +4411,14 @@ private void writeAESTimeRange (String baseIndent,
...
+    /**
+     * Clean up a URI string by escaping forbidden characters. 
      * We assume (perhaps dangerously) that a % is the start of
-     * an already escaped hexadecimal sequence. */
-    private String cleanURIString (String uri) {
+     * an already escaped hexadecimal sequence.
+     * @param uri
+     * @return
+     */
+    public String cleanURIString (String uri) {
```
Comment (tledoux) [inline jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java:4421]: Shouldn't this method be static ?

**Review by tledoux — COMMENTED** [review]
My feeling is that the 4 methods (encodeContent, encodeValue, isPropertyEmpty, cleanURIString) are really utility functions. It would be easier for maintenance to put them all in a Utility class, so we know others may use them and we don't mess with them.

File: jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java (hunk 2 above)
Comment (carlwilson) [reply jhove-core/src/main/java/edu/harvard/hul/ois/jhove/handler/XmlHandler.java]: I'd not thought that clearly about it before @tledoux but I think you're right. Will pr a refactor.

**Comment by carlwilson** [comment]
I'd not thought that clearly about it before @tledoux but I think you're right. Will pr a refactor.

**Comment by carlwilson** [comment]
Included in #538

**Pull request closed without merging**